desc/objectives/_free_boundary.py

```diff
@@ -754,6 +754,341 @@ def _print(fmt, fmax, fmin, fmean, norm, units):
             _print(fmt, fmax, fmin, fmean, norm, unit)
 
 
+class QuadraticFlux(_Objective):
+    """Target B*n = 0 on LCFS.
+
+    Uses virtual casing to find plasma component of B and penalizes
+    (B_coil + B_plasma)*n
+
+    Parameters
+    ----------
+    ext_field : MagneticField
+        External field produced by coils.
+    eq : Equilibrium, optional
+        Equilibrium that will be optimized to satisfy the Objective.
+    target : float, ndarray, optional
+        Target value(s) of the objective. Only used if bounds is None.
+        len(target) must be equal to Objective.dim_f
+    bounds : tuple, optional
+        Lower and upper bounds on the objective. Overrides target.
+        len(bounds[0]) and len(bounds[1]) must be equal to Objective.dim_f
+    weight : float, ndarray, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        len(weight) must be equal to Objective.dim_f
+    normalize : bool
+        Whether to compute the error in physical units or non-dimensionalize.
+    normalize_target : bool
+        Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True.
+    s, q : integer
+        Hyperparameters for singular integration scheme, s is roughly equal to the size
+        of the local singular grid with respect to the global grid, q is the order of
+        integration on the local grid
+    src_grid : Grid, optional
+        Collocation grid containing the nodes for plasma source terms.
+    eval_grid : Grid, optional
+        Collocation grid containing the nodes on the plasma surface at which the
+        magnetic field is being calculated and where to evaluate Bn errors.
+    field_grid : Grid, optional
+        Grid used to discretize ext_field (e.g. grid for the magnetic field source from
+        coils).
+    eq_fixed : bool
+        Whether or not the equilibrium's DOF (params) change during optimization.
+        If true, Bplasma will be precomputed in build().
+    field_fixed : bool
+        Whether or not the external field's DOF (params) change during optimization.
+    vacuum : bool
+        If true, Bplasma is set to zero.
+    loop : bool
```

Comment:
if we're only evaluating the integral once I think we can just fix `loop=True` to avoid memory issues